benchmark_runner/workloads/vdbench_vm.py

```diff
@@ -40,32 +63,62 @@ def run(self):
-            # Create vm log should be direct after vm is ready
-            self.__vm_name = self._create_vm_log(labels=[self.__workload_name])
-            self.__status = self._oc.wait_for_vm_log_completed(vm_name=self.__vm_name, end_stamp=self.END_STAMP)
-            self.__status = 'complete' if self.__status else 'failed'
-            # save run artifacts logs
-            result_list = self._create_vm_run_artifacts(vm_name=self.__vm_name, start_stamp=self.START_STAMP, end_stamp=self.END_STAMP)
-            if self._es_host:
-                # upload several run results
-                for result in result_list:
-                    self._upload_to_elasticsearch(index=self.__es_index, kind=self.__kind, status=self.__status, result=result)
-                # verify that data upload to elastic search according to unique uuid
-                self._verify_elasticsearch_data_uploaded(index=self.__es_index, uuid=self._uuid)
-            self._oc.delete_vm_sync(
-                yaml=os.path.join(f'{self._run_artifacts_path}', f'{self.__name}.yaml'),
-                vm_name=self.__vm_name)
+            if not self._scale:
+                self._oc.create_vm_sync(yaml=os.path.join(f'{self._run_artifacts_path}', f'{self.__name}.yaml'), vm_name=self.__vm_name)
+                self._oc.wait_for_ready(label=f'app=vdbench-{self._trunc_uuid}', run_type='vm', label_uuid=False)
+                # Create vm log should be direct after vm is ready
+                self.__vm_name = self._create_vm_log(labels=[self.__workload_name])
+                self.__status = self._oc.wait_for_vm_log_completed(vm_name=self.__vm_name, end_stamp=self.END_STAMP)
+                self.__status = 'complete' if self.__status else 'failed'
+                # save run artifacts logs
+                result_list = self._create_vm_run_artifacts(vm_name=self.__vm_name, start_stamp=self.START_STAMP, end_stamp=self.END_STAMP)
+                if self._es_host:
+                    # upload several run results
+                    for result in result_list:
+                        self._upload_to_elasticsearch(index=self.__es_index, kind=self.__kind, status=self.__status, result=result)
+                    # verify that data upload to elastic search according to unique uuid
+                    self._verify_elasticsearch_data_uploaded(index=self.__es_index, uuid=self._uuid)
+                self._oc.delete_vm_sync(
+                    yaml=os.path.join(f'{self._run_artifacts_path}', f'{self.__name}.yaml'),
+                    vm_name=self.__vm_name)
+            # scale
+            else:
+                # create redis and state signals
+                sync_pods = {'redis': 'redis', 'state_signals_exporter_pod': 'state-signals-exporter'}
+                for pod, name in sync_pods.items():
+                    if pod == 'redis':
+                        pod_name = f'{name}-master'
+                    else:
+                        pod_name = name
```

Comment:
See previous comment about naming; just use redis-master as the name.

Reply:
Fixed